src/xml_parse.rs

```diff
@@ -9,13 +9,39 @@ use xml::reader::{EventReader, XmlEvent};
 
 use super::db::{AutoType, AutoTypeAssociation, Entry, Group, Value};
 
+extern crate chrono;
+
 #[derive(Debug)]
 enum Node {
     Entry(Entry),
     Group(Group),
     KeyValue(String, Value),
     AutoType(AutoType),
     AutoTypeAssociation(AutoTypeAssociation),
+    ExpiryTime(String),
+    Expires(bool),
+}
+
+fn parse_xml_timestamp(t: &str) -> Result<chrono::NaiveDateTime> {
+    match chrono::NaiveDateTime::parse_from_str(t, "%Y-%m-%dT%H:%M:%SZ") {
```

Comment:
In ISO 8601, [`Z` suffix is for specifying "Zulu" time](https://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators). Can you use [`chrono::DateTime::parse_from_rfc3339`](https://docs.rs/chrono/0.4.0/chrono/struct.DateTime.html#method.parse_from_rfc3339) instead?

Reply:
I am very happy to have a second set of eyes and sounding board for the approach you see here.

Background on how I arrived at this design:

KeePassXC (and, I believe the original KeePass itself) stores timestamps, at least the ExpiryTime, with the Z timezone indicator.  But, what they are actually storing is the local machine time not converted to Zulu/UTC time.  This is neither ISO 8601 compliant (where the timezone is optional) nor RFC 3339 compliant (where a timezone is required and intended to be accurate).  Sadly, I have seen this approach more times than I would like to admit; so, it did not surprise me.

I tested this with KeePassXC on a machine with the time zone set to CDT (UTC−05:00).  The ExpiryTime was stored with the Z timezone indicator but not actually converted to UTC.  

chrono::DateTime::parse_from_rfc3339 returns a DateTime<FixedOffset>, this would accurately represent what is stored in the XML document text but might mislead anyone using this crate.  

chrono::NaiveDateTime::parse_from_str returns a NaiveDateTime which seemed to me a better representation of these timestamps for users of this crate.  (NaiveDateTime does not have a parse_from_rfc3339 method.)

Please let me know how you want me to proceed with this:
1) Use chrono::DateTime in this crate (and ::parse_from_rfc3339 here) instead of chrono::NaiveDateTime
2) Stick with the current design and code
3) More testing (other KeePass clients, other timestamp fields such as Created and Modified, etc.)
4) Comment reasons for this approach more thoroughly
5) Other (If so, please elaborate.)

Reply:
Oh wow, that is a very footgun-style design. In this case I fully agree with your decision and would stick with your code as-is. I would only request some additional docs for the appropriate fields in the `Entry` struct [so that they show up in the generated docs](https://docs.rs/keepass/0.4.7/keepass/struct.Entry.html) (i.e. with a `///` comment)

Reply:
As much as I wanted to include footgun-style design in those comments, I decided something a bit gentler might be better for publishing.